source/common/http/headers.h

```diff
@@ -133,6 +133,7 @@ class HeaderValues {
   const LowerCaseString Cookie{"cookie"};
   const LowerCaseString Date{"date"};
   const LowerCaseString EnvoyAttemptCount{absl::StrCat(prefix(), "-attempt-count")};
+  const LowerCaseString EnvoyAuthHeadersToRemove{absl::StrCat(prefix(), "-auth-headers-to-remove")};
   const LowerCaseString EnvoyAuthPartialBody{absl::StrCat(prefix(), "-auth-partial-body")};
```

Comment:
Can you move these to be defined inside the extauth filter as these are extension specific? Also, where are these magic headers documented? It wasn't clear to me from the docs how I would use these.

Reply:
@mattklein123 I've moved the two auth headers to a new class in the ext_authz namespace that mimicks how we do things in `headers.h`. I hope that looks okay.

When it comes to documentation the use cases for these headers are a bit obscure. `x-envoy-auth-partial-body` is only used when `with_request_body` is set, and as a result (I imagine) it's only mentioned in that proto field docstring: https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/ext_authz/v3/ext_authz.proto

The new `x-envoy-auth-headers-to-remove` is only used by the http auth server when it's sending a response. It will appear under the new proto field `headers_to_remove` once the PR is merged: https://www.envoyproxy.io/docs/envoy/latest/api-v3/service/auth/v3/external_auth.proto.html?highlight=checkrequest#envoy-v3-api-msg-service-auth-v3-okhttpresponse

Do you think these headers should be mentioned in a more central location?

Reply:
> The new x-envoy-auth-headers-to-remove is only used by the http auth server when it's sending a response. It will appear under the new proto field headers_to_remove once the PR is merged:

Apologies, I missed this. LGTM!